### PR TITLE
Update actions/setup-python@v2 -> v4

### PR DESCRIPTION
**Pull request by Kaushik-Iyer**
## Motivation
Resolve a part of #4165 
## Description of the changes
update action/setup-python@v2 -> v4 in all workflows.

### CODE REVIEW

**Comment by Kaushik-Iyer** [comment]
I don't understand what has gone wrong

**Comment by nzw0301** [comment]
I could reproduce the failed case on my mac:

```bash
mpirun -n 2 -- pytest tests/integration_tests/test_pytorch_distributed.py::test_should_prune
```

I suppose we use mpirun, [`storage_specifier`](https://github.com/optuna/optuna/blob/master/optuna/testing/storages.py#L56) is `ParameterSet` defined by https://github.com/optuna/optuna/blob/master/optuna/testing/storages.py#L32-L41, so if block cannot capture this for dask.

**Comment by nzw0301** [comment]
I think this PR will do nothing about this issue. Another PR should address it.

**Comment by nzw0301** [comment]
Ah, https://github.com/optuna/optuna/pull/4301 will fix this issue on `mac-tests`.

File: .github/workflows/mac-tests.yml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
       uses: actions/checkout@v3
 
     - name: Setup Python3.8
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
```
Comment (Alnusjaponica) [inline .github/workflows/mac-tests.yml:31]:
```suggestion
      uses: actions/setup-python@v2
```

File: .github/workflows/mac-tests.yml [hunk 2]
```diff
@@ -67,7 +67,7 @@ jobs:
...
```
Comment (Alnusjaponica) [inline .github/workflows/mac-tests.yml:70]:
```suggestion
      uses: actions/setup-python@v2
```

**Comment by Alnusjaponica** [comment]
You can just leave `mac-tests.yml` unchanged to escape `mac-tests/tests-integration-mac` as in #4207.

**Comment by toshihikoyanase** [comment]
@Kaushik-Iyer  The PR was marked as `stale`. I think we can make it review-ready after applying @Alnusjaponica 's suggestions. So, let me apply them.

**Comment by Kaushik-Iyer** [comment]
@toshihikoyanase is everything fine now?

**Comment by toshihikoyanase** [comment]
@Kaushik-Iyer Yes, I think the PR is review-ready.

@c-bata You may be busy, but could you assign reviewers, please?
I guess @Alnusjaponica may be interested in this PR.

**Comment by c-bata** [comment]
@Kaushik-Iyer I'm sorry that I missed this PR being ready for review :bow:

Let me assign @Alnusjaponica  and @contramundum53.

**Comment by Alnusjaponica** [comment]
I read through the [Release note](https://github.com/actions/setup-python/releases?page=1). It seems that they just fixed minor bugs, updated dependency, added some new features, and therefore made no significant changes which affect our CI.

**Review by Alnusjaponica — APPROVED** [review]

**Comment by c-bata** [comment]
@contramundum53 Could you please close #4165 after merging this PR.

**Comment by contramundum53** [comment]
LGTM.

**Review by contramundum53 — APPROVED** [review]

**Pull request merged**